packages/core/src/amazonqFeatureDev/errors.ts

```diff
@@ -3,140 +3,163 @@
     constructor(message: string, code: string) {
         super(message, { code })
     }
 }
 
-export class PrepareRepoFailedError extends ToolkitError {
+export class PrepareRepoFailedError extends ServiceError {
     constructor() {
         super(i18n('AWS.amazonq.featureDev.error.prepareRepoFailedError'), {
             code: 'PrepareRepoFailed',
         })
     }
 }
 
-export class UploadCodeError extends ToolkitError {
+export class UploadCodeError extends ServiceError {
     constructor(statusCode: string) {
         super(uploadCodeError, { code: `UploadCode-${statusCode}` })
     }
 }
 
-export class UploadURLExpired extends ToolkitError {
+export class UploadURLExpired extends ClientError {
     constructor() {
         super(i18n('AWS.amazonq.featureDev.error.uploadURLExpired'), { code: 'UploadURLExpired' })
     }
 }
 
-export class IllegalStateTransition extends ToolkitError {
+export class IllegalStateTransition extends ServiceError {
     constructor() {
         super(i18n('AWS.amazonq.featureDev.error.illegalStateTransition'), { code: 'IllegalStateTransition' })
     }
 }
 
-export class ContentLengthError extends ToolkitError {
+export class IllegalStateError extends ServiceError {
+    constructor(message: string) {
+        super(message, { code: 'IllegalStateTransition' })
+    }
+}
+
+export class ContentLengthError extends CommonContentLengthError {
     constructor() {
         super(i18n('AWS.amazonq.featureDev.error.contentLengthError'), { code: ContentLengthError.name })
     }
 }
 
-export class ZipFileError extends ToolkitError {
+export class ZipFileError extends ServiceError {
     constructor() {
         super(i18n('AWS.amazonq.featureDev.error.zipFileError'), { code: ZipFileError.name })
     }
 }
 
-export class CodeIterationLimitError extends ToolkitError {
+export class CodeIterationLimitError extends ClientError {
     constructor() {
         super(i18n('AWS.amazonq.featureDev.error.codeIterationLimitError'), { code: CodeIterationLimitError.name })
     }
 }
 
-export class MonthlyConversationLimitError extends ToolkitError {
+export class MonthlyConversationLimitError extends ClientError {
     constructor(message: string) {
         super(message, { code: MonthlyConversationLimitError.name })
     }
 }
 
-export class UnknownApiError extends ToolkitError {
+export class UnknownApiError extends ServiceError {
     constructor(message: string, api: string) {
         super(message, { code: `${api}-Unknown` })
     }
 }
 
-export class ApiError extends ToolkitError {
+export class ApiClientError extends ClientError {
+    constructor(message: string, api: string, errorName: string, errorCode: number) {
+        super(message, { code: `${api}-${errorName}-${errorCode}` })
+    }
+}
+
+export class ApiServiceError extends ServiceError {
     constructor(message: string, api: string, errorName: string, errorCode: number) {
         super(message, { code: `${api}-${errorName}-${errorCode}` })
```

Comment:
What is the difference between ServiceError/ApiServiceError/ApiError ? Seems very unintuitive.